integration_tests/test_parser.py

```diff
@@ -15,64 +15,76 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 import os
+import fixtures
 import subprocess
 
 import integration_tests
+import testscenarios
 from snapcraft.tests import fixture_setup
 
 
-class TestParser(integration_tests.TestCase):
+class ParserTestCase(integration_tests.TestCase):
     """Test bin/snapcraft-parser"""
 
-    def test_parser_basic(self):
-        """Test snapcraft-parser basic usage"""
-        fixture = fixture_setup.FakePartsWiki()
-        self.useFixture(fixture)
+    def setUp(self):
+        super().setUp()
 
-        args = [self.snapcraft_parser_command, '--index',
-                fixture.fake_parts_wiki_fixture.url,
-                '--debug', '--output', 'parts.yaml']
-        subprocess.check_call(args, stderr=subprocess.DEVNULL,
-                              stdout=subprocess.DEVNULL)
+        tempdir = fixtures.TempDir()
+        self.useFixture(tempdir)
+        self.useFixture(fixtures.EnvironmentVariable('TMPDIR', tempdir.path))
+        self.tempdir = tempdir.path
 
-        self.assertTrue(os.path.exists('parts.yaml'))
+    def call_parser(self, wiki_path, expect_valid, expect_output=True):
+        part_file = os.path.join(self.tempdir, 'parts.yaml')
+        args = [self.snapcraft_parser_command, '--index', wiki_path,
+                '--debug',
+                '--output', part_file]
```

Comment:
I would move this to the parent class, because it should be almost the same as run_snapcraft which is defined there. And I would call it run_snapcraft_parser. In fact, I'm doing it here:
https://github.com/snapcore/snapcraft/pull/917/files#diff-e0600b57bbf57ae4d401975ecad5aca2R83

Your branch will land earlier than mine probably.